src/mainviews/create-job.tsx

```diff
@@ -89,10 +89,22 @@ export function CreateJob(props: ICreateJobProps): JSX.Element {
       const envList = await api.getRuntimeEnvironments();
       setEnvironmentList(envList);
       if (envList.length === 1) {
+        // If no default compute type is specified, show the first one by default
+        let newComputeType = envList[0].compute_types?.[0];
+
+        // Validate that the default compute type is in fact in the list
+        if (
+          envList[0].default_compute_type &&
+          envList[0].compute_types &&
+          envList[0].compute_types.indexOf(envList[0].default_compute_type) >= 0
```

Comment:
`envList[0].compute_types.includes(envList[0].default_compute_type)` could be used here. It would save just 3 characters (no >= 0 needed) but might improve readability. At your discretion